pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -564,16 +564,16 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 		return false
 	}
 
-	// Allocates from reusableDevices list first.
-	if allocateRemainingFrom(reusableDevices) {
-		return allocated, nil
-	}
-
 	// Needs to allocate additional devices.
 	if m.allocatedDevices[resource] == nil {
 		m.allocatedDevices[resource] = sets.NewString()
 	}
 
+	// Allocates from reusableDevices list first.
```

Comment:
The functor is reused in other places. I don't think we should change the logic here since the functor is keeping a count. This is the simplest change to prevent the crash.